lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -938,6 +952,11 @@ void TFFunctionPartition::diagnoseCopyToAccelerator(
 
   // Try to determine a good source location to report.
   auto loc = getUserSourceLocation(value);
+  
+  // Opaque handles can never be sent or passed as tenosr program arguments.
+  // Deabstraction must have rejected this earlier.
```

Comment:
how does deabstraction know if this needs to be a TF->swift send? That is computed in the partition pass.

Also in `diagnoseUsesFromHost()` below, we use if check and emit diagnostics -- should we do the same here instead of assert failure?

Reply:
This a Swift->TF send, not the other way around. I can't think of a scenario where an opaque will be sent to the accelerator, because **public functions** and **`@inline(never)` functions** can never return an opaque handle unless they are `@convention(tensorflow)`. This should be diagnosed by type checking.

Since there can't be a case where an opaque handle is a) passed to a host function, or b) returned by an `@inline(never)` function, partitioning will never encounter a case where an opaque handle gets sent from host to the accelerator.

Reply:
Can I write a non-@convention(tensorflow) public function today that creates a returns an opaque handle? If that's already rejected, I might have missed reading some code...

Overall the reasoning looks solid. Thanks! Can you add it as a code comment to capture that piece of knowledge?